File: src/dataset/javascript/pascals-triangle_exercism-javascript-PhymasSC/claude/ClaudeSonnet4_pascals-triangle_v2.js
//
// This is only a SKELETON file for the 'Pascals Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const rows = number => {
	if (number <= 0) return [];
	let res = new Array(number);
	
	for (let row = 0; row < number; row++) {
		res[row] = new Array(row + 1);
		res[row][0] = 1;
		res[row][row] = 1;
		
		for (let column = 1; column < row; column++) {
			res[row][column] = res[row - 1][column - 1] + res[row - 1][column];
		}
	}
	return res;
};

File: src/dataset/go/pascals_triangle_Exercism_go_drapala/claude/ClaudeSonnet4_pascals_triangle_v2.go
package pascal

func Triangle(n int) [][]int {
	if n == 0 {
		return [][]int{}
	}
	
	result := make([][]int, n)
	
	for i := 0; i < n; i++ {
		row := make([]int, i+1)
		row[0] = 1
		row[i] = 1
		
		for j := 1; j < i; j++ {
			row[j] = result[i-1][j-1] + result[i-1][j]
		}
		
		result[i] = row
	}
	
	return result
}